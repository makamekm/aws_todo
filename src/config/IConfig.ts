import * as ORM from "typeorm";
import {
    LogLevel,
} from "../log";

export interface ITypeORMConfigScope {
    [name: string]: ORM.ConnectionOptions;
}

export interface IAdditionalInformation {
    [name: string]: (IAdditionalInformation | string | number | boolean);
}

export interface IRedisConfigScope {
    [name: string]: {
        port: number;
        host: string;
        password?: string;
        scope?: string;
    };
}

export interface IConfigScope {
    [name: string]: IConfig;
}

export interface IConfig {
    // Log to console with level: [debug, info, warn, error] ["info", "warn", "debug", "error"]
    logLevel?: LogLevel;
    // additional log messages to console [string[]]
    additionalLogMessages?: string[];
    // ORM via typeorm [scope: typeorm object]
    db?: ITypeORMConfigScope;
    // client config in state will be available in browser store [json object]
    publicConfig?: any;
}
