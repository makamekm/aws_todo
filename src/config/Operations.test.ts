import { IConfigScope } from "./IConfig";
import {
    applyDefaultConfig,
    getConfig,
    readConfig,
    saveConfig,
    setConfig,
} from "./Operations";

describe("Config", () => {
    describe("Operations", () => {
        afterEach(() => {
            applyDefaultConfig();
        });

        beforeEach(() => {
            applyDefaultConfig();
        });

        it("should getConfig match snapshot", () => {
            expect(getConfig()).toMatchSnapshot();
        });

        it("should readConfig from a file", () => {
            const conf = {
                default: {
                    db: "test",
                    logLevel: "test",
                },
            };
            readConfig(<any> {
                existsSync: jest.fn().mockReturnValue(true),
                readFileSync: jest.fn().mockReturnValue(
                    JSON.stringify(conf),
                ),
            });
            expect(getConfig()).toMatchObject(conf.default);
        });

        it("should saveConfig save config into a file", () => {
            const spy = jest.fn();
            saveConfig(<any> {
                existsSync: jest.fn().mockReturnValue(true),
                writeFileSync: spy,
            });
            expect(spy).toBeCalled();
        });

        it("should setConfig match object", () => {
            const conf: IConfigScope = {
                default: {
                    db: {},
                    logLevel: "debug",
                },
            };
            setConfig(conf);
            expect(getConfig()).toMatchObject(conf.default);
        });
    });
});
