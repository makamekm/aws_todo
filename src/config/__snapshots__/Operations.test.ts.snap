// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Config Operations should getConfig match snapshot 1`] = `
Object {
  "db": Object {
    "default": Object {
      "database": "todo",
      "host": "localhost",
      "password": "",
      "port": 5432,
      "synchronize": true,
      "type": "postgres",
      "username": "root",
    },
  },
  "logLevel": "info",
  "publicConfig": Object {
    "graphqlEndpoint": "http://localhost:3000/graphql",
  },
}
`;
